bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java

```diff
@@ -598,21 +597,15 @@ private void scheduleReconnectJob() {
             if (!connector.isConnected()) {
                 logger.debug("Trying to reconnect...");
                 closeConnection();
-                String error = null;
                 if (openConnection()) {
                     logger.debug("Reconnected");
                     // Polling status will disconnect from MPS4 on reconnect
                     if (!isMps4) {
                         pollStatus();
                     }
                 } else {
-                    error = "Reconnection failed";
-                }
-                if (error != null) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Reconnection failed");
                     closeConnection();
-                } else {
-                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, this.versionString);
```

Comment:
You want now to set the thing status ONLINE only when something is received through the connection ?
In practice, will a message be received systematically quickly after the connection ?

Reply:
Yes, for serial port connections, pollStatus() will cause messages to be received almost immediately. For MPS4, a ping message will arrive within a few seconds.

Reply:
Ok